shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -573,14 +697,39 @@ - (void)insertText:(id)string replacementRange:(NSRange)range {
 }
 
 - (void)doCommandBySelector:(SEL)selector {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wundeclared-selector"
+  // For key sequences that aren't mapped to anything, the
+  // macOS keybinding manager will send us a "noop:" selector.
+  if (sel_isEqual(selector, @selector(noop:))) {
+    return;
+  }
+#pragma clang diagnostic pop
+
+  for (SelectorIntent* intent in kIntentLookUp) {
```

Comment:
Could you make kIntentLookup an NSSet to make this lookup faster?

Reply:
selectors are not hashable and need a special global function to compare equality (`sel_isEqual`).